server/vulnerabilities/nvd/cve_test.go

```diff
@@ -660,23 +671,24 @@ func TestPreprocessVersion(t *testing.T) {
 		input    string
 		expected string
 	}{
-		{"2.3.0.2", "2.3.0+2"},
-		{"2.3.0+2", "2.3.0+2"},
-		{"v2.3.0+2", "v2.3.0+2"},
-		{"2.3.0.2.5", "2.3.0+2.5"},
+		{"2.3.0.2", "2.3.0-2"},
+		{"2.3.0+2", "2.3.0-2"},
+		{"v5.3.0+2", "v5.3.0-2"},
+		{"5.3.0-2", "5.3.0-2"},
+		{"2.3.0.2.5", "2.3.0-2.5"},
 		{"2.3.0", "2.3.0"},
 		{"2.3", "2.3"},
 		{"v2.3.0", "v2.3.0"},
 		{"notAVersion", "notAVersion"},
-		{"2.0.0+svn315-7fakesync1ubuntu0.22.04.1", "2.0.0+svn315-7fakesync1ubuntu0.22.04.1"},
```

Comment:
We should still have:
`{"2.0.0+svn315-7fakesync1ubuntu0.22.04.1", "2.0.0-svn315-7fakesync1ubuntu0.22.04.1"},`
right?

Reply:
good catch, i reverted that test

Reply:
Sorry, I'm still a bit confused.

So `2.3.0+2` is a valid version according to https://semver.org/ but the [semver](https://github.com/Masterminds/semver) package does not support it?

Reply:
may be a bug in the `semver` package. check this out: https://go.dev/play/p/DsOOiVwG_l5

Reply:
Ah makes sense (from https://semver.org/) - build is whatever is after `+`
```
Build metadata MUST be ignored when determining version precedence. Thus two versions that differ only in the build metadata, have the same precedence.
```

So am ok with transforming `.` to `-`, but should we transform `+` to `-`? (`+` means build, `-` is pre-release which is comparable.)

Reply:
Good call, i removed the `+` to `-` transformation